cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorNonParametrizedTest.java

```diff
@@ -137,7 +138,8 @@ public void testCustomLabelsAndAnnotations(VertxTestContext context) {
         when(podOps.listAsync(eq(NAMESPACE), any(Labels.class))).thenReturn(Future.succeededFuture(List.of()));
 
         KafkaAssemblyOperator op = new KafkaAssemblyOperator(vertx, new PlatformFeaturesAvailability(false, KubernetesVersion.V1_16), certManager, passwordGenerator,
-                supplier, ResourceUtils.dummyClusterOperatorConfig(1L));
+                supplier, ResourceUtils.dummyClusterOperatorConfig(1L),
+                Mockito.mock(KubernetesRestartEventPublisher.class));
```

Comment:
Just FYI: These are all the things we would get for free if it is part of the supplier.

Reply:
:+1: